code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -0,0 +1,110 @@
+//I JUST WANNA GRILL FOR GOD'S SAKE
+
+/obj/machinery/grill
+	name = "grill"
+	desc = "Just like the old days."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "grill_open"
+	density = TRUE
+	layer = BELOW_OBJ_LAYER
+	use_power = NO_POWER_USE
+	var/grill_fuel = 0
+	var/grilled_item
+	var/datum/looping_sound/grill/grill_loop
+
+/obj/machinery/grill/Initialize()
+	..()
+	grill_loop = new(list(src), FALSE)
+
+/obj/machinery/grill/update_icon()
+	if(grilled_item)
+		icon_state = "grill"
+	else if(!grill_fuel)
+		icon_state = "grill_open"
+	else
+		icon_state = "grill_on"
```

Comment:
Why is the default state grill on

Reply:
default state is grill_open, grill on if it has fuel